acme/acme/client_test.py

```diff
@@ -127,6 +127,13 @@ def test_query_registration(self):
         self.response.json.return_value = self.regr.body.to_json()
         self.assertEqual(self.regr, self.client.query_registration(self.regr))
 
+    def test_query_registration_updates_new_authzr_uri(self):
```

Comment:
I think this test would hold true without changes from this PR. What you want to test is the `_regr_from_response(response, new_authzr_uri='a').new_authzr_uri == 'UPDATED'` case. Or maybe you want to keep the test and revert changes to the `_regr_from_response`?